src/codecoverage/bot/code_coverage_bot/zero_coverage.py

```diff
@@ -85,7 +86,51 @@ def get_fileinfo(self, filenames):
 
         return res
 
-    def generate(self, artifacts, hgrev, gitrev, out_dir='.'):
+    def generate_zero_coverage_some_platform_only(self, report, hgrev, gitrev, out_dir='.'):
+        '''
+        :param report: dictionary with key platform name, and value is report generated
+                       using ZeroCov.generate for that platform
+        :param hgrev: hg revision, must be same for all reports
+        :param gitrev: git revision, must be same for all reports
+        :param out_dir: directory to output the json result
+        '''
+
+        platforms = ['linux', 'windows']
+
+        git_revision = set([report[platform]['github_revision'] for platform in platforms])
+        hg_revision = set([report[platform]['hg_revision'] for platform in platforms])
+        if len(git_revision) != 1 and len(hg_revision) != 1:
+            raise Exception('Git revision and Hg revision must be the same')
+
+        uncovered_file_name = collections.defaultdict(lambda: set())
+        all_files = {}
+
+        for platform in platforms:
+            for file in report[platform]['files']:
+                if file['uncovered'] is False:
+                    continue
+                uncovered_file_name[file['name']].add(platform)
+
+                all_files[file['name']] = file
+
+        zero_coverage_info = []
+        for fname, info in all_files.items():
+            covered = list(set(platforms) - uncovered_file_name[fname])
+            uncovered = list(uncovered_file_name[fname])
+
+            info.update({'covered_platform': covered,
+                         'uncovered_platform': uncovered})
+            zero_coverage_info.append(info)
+
+        zero_coverage_report = {'github_revision': gitrev,
+                                'hg_revision': hgrev,
+                                'files': zero_coverage_info}
+
+        os.makedirs(os.path.join(out_dir, 'zero_coverage_some_platform_functions'), exist_ok=True)
+        with open(os.path.join(out_dir, 'zero_coverage_some_platform_functions_report.json'), 'w') as f:
+            json.dump(zero_coverage_report, f)
+
+    def generate(self, artifacts, hgrev, gitrev, out_dir='.', return_result=False):
```

Comment:
Nit: `writeout_result` or something like this, instead of `return_result`. Or, you could also make `generate` always return the result with the `return` statement, and then the caller of `generate` write the result in the file.